chainer/link.py

```diff
@@ -205,14 +266,36 @@ def add_persistent(self, name, value):
                 for the attribute name.
             value: Value to be registered.
 
+        Returns:
+            The registered value.
+
         """
         d = self.__dict__
         if name in d:
             raise AttributeError(
                 'cannot register a new persistent value %s: attribute exists'
                 % name)
-        self._persistent.append(name)
+        self._persistent.add(name)
+        self._params.discard(name)
         d[name] = value
+        return value
+
+    def register_persistent(self, name):
```

Comment:
Yes. I did not make it deprecated because the persistent value support is not mature (it is currently only used by `BatchNormalization` and `SimplifiedDropConnect` AFAIK).